examples/allocation-endpoint/README.md

```diff
@@ -0,0 +1,104 @@
+# Allocation Endpoint
+
+This is a sample solution to enable an Allocation Endpoint proxy in front of GKE Agones clusters.
+
+In this folder you will find:
+
+1. Terraform modules that creates resources in GCP
+2. The Allocation Endpoint Proxy code that gets deployed to Cloud Run
+3. An ESP container that gets deployed to agones-allocator
+4. A client sample code to send allocation requests to the proxy
+5. Dockerfile that builds the proxy and scripts to push the image to a docker repository
+6. Documentation on how to use the solution.
+
+Here is the architecture of GCP resources created:
+
+![architecture](https://github.com/googleforgames/agones/blob/main/examples/allocation-endpoint/architecture.png?raw=true)
+
+
+## GKE cluster
+First and foremost you need to create clusters and installing Agones for experimenting with this solution.
+The clusters can be in the same GCP project as your Allocation Endpoint proxy or it can be in different project.
+
+---
+**NOTE**
+
+The solution is not experimented with non-GKE clusters.
+
+---
+
+When creating GKE Clusters, [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) needs to be enabled.
+
+<pre>
+gcloud container clusters create [NAME] --cluster-version=1.21 \
+  --project=[GKE-PROJECT-ID] \
+  --tags=game-server \
+  --scopes=gke-default \
+  --num-nodes=4 \
+  --machine-type=e2-standard-4 \
+  <b>--workload-pool=[GKE-PROJECT-ID].svc.id.goog \ </b>
+  --region=[REGION]
+</pre>
+
+Install Agones on your cluster. You need to disable mTLS because the `agones-allocator` container will be the backend for ESP container.
+
+```
+helm upgrade my-release --install --namespace agones-system --create-namespace agones/agones \
+  --set agones.allocator.disableMTLS=true \
+  --set agones.allocator.disableTLS=true \
+  --set agones.allocator.service.http.enabled=false
+```
+
+After installing Agones, deploy [ESP](https://cloud.google.com/endpoints/docs/grpc/specify-esp-v2-startup-options) which is an envoy based proxy in front of agones `agones-alloator` container. Run the following to patch the service deployement, change the service port to ESP and add annotation to `agones-allocator` service account to impersonate GCP service account. 
+
+Replace [GKE-PROJECT-ID] in `patch-agones-allocator.yaml` with your project ID before runing the scripts.
```

Comment:
Maybe a future improvement, but this could be done via `sed` in the following commands if the project is set an an environment variable.

Reply:
I like `envsubst` better for something like this. Just noting for maybe future enhancement.

Actually, what would be fun - use the environment variable that is usually set in cloud shell for the GCP  PROJECT - saves a lot of time.

Or we could also cloud build the entire thing.